### ISSUE TITLE
Protocol relative URLs considered local by Phalcon

### ISSUE THREAD

**Issue by phresnel** [phalcon/cphalcon#10236]
In https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/url.zep#L183 , only urls containing a colon (`:`) have the chance of being considered as non-local urls.

This may fail when you have a local url with a port (not tested, though), and fails if you have an absolute, but protocol relative url.

For example, if you have in your Volt-file something like

    form("//example.com/frob/edit")

then `Url\get` will wrongly transform this to `//example.com///example.com/frob/edit`, given `//example.com` the base path.

I could fix this myself, including a backport to version 1.3.5, however, I fail to find documentation about the developer workflow. Given the latter, for both 2.0 and 1.3.5, I'd be happy to provide a patch.

**Comment by dreamsxin** [comment]
See https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md#contributions

**Comment by phresnel** [comment]
Yes, but for example, I couldn't find any sources in the 1.3.5 branch. How would I create a patch for 1.3.5?

**Comment by dreamsxin** [comment]
See https://github.com/phalcon/cphalcon/tree/1.3.5
Suggestions work in the v2.0 https://github.com/phalcon/cphalcon/tree/2.0.x

**Comment by phresnel** [comment]
Yes, but the only sources in the 1.3.5 branch are generated C files. And even if you hack on those, neither `./install`, nor `./make` will honor your changes.

**Comment by andresgutierrez** [comment]
@phresnel You can to compile from ext/

**Comment by phresnel** [comment]
@andresgutierrez Yes, yes, yes. But those are not the source codes, but generated C files.

I guess Phalcon has no real interest in contributions, and I lost it, too.

**Comment by dreamsxin** [comment]
@phresnel The Phalcon2, You change code in https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/url.zep#L180
And, you can use this generated c files and make it:
```shell
zephir generate
cd ext
make && make install
```
See version:
```shell
php --ri phalcon
```

**Comment by phresnel** [comment]
Phalcon 2 is _not_ the problem ...

> Yes, but for example, I couldn't find any sources in the 1.3.5 branch. How would I create a patch for 1.3.5?

**Comment by andresgutierrez** [comment]
@phresnel Phalcon 1.3.x is written in C, you have to submit a patch in C

**Comment by phresnel** [comment]
Okay, _that's_ an answer. Tbh, the code of Phalcon 1 looked pretty non-idiomatic, which is why I wanted to be sure.

Thanks. Yet I am not sure if I'll find the time again to work on this.

**Comment by andresgutierrez** [comment]
This is fixed in Phalcon 2.0.4 (branch 2.0.x)

**Comment by phresnel** [comment]
Neat, thanks for taking a look
